src/blueprint/blueprint.rs

```diff
@@ -219,4 +221,61 @@ impl Blueprint {
     // generation of schema cannot fail.
     schema.finish().unwrap()
   }
+
+  /// Generate a dumb schema useful for validation.
+  pub fn to_validation_schema(&self) -> Result<Schema, ValidationError<String>> {
+    match create(self, Some(SchemaModifiers { no_resolver: true }))
+      .validation_mode(ValidationMode::Strict)
+      .disable_introspection()
+      .finish()
+    {
+      Ok(schema) => Ok(schema),
+      Err(e) => Err(ValidationError::new(e.to_string())),
+    }
+  }
+
+  pub async fn validate_operation(schema: &Schema, query: &str) -> Vec<Cause<String>> {
+    schema
+      .execute(query)
+      .await
+      .errors
+      .iter()
+      .map(|err| {
+        Cause::new(err.message.clone()).trace(
+          err
+            .locations
+            .iter()
+            .map(|loc| format!("line {} column {}", loc.line, loc.column))
+            .collect(),
+        )
+      })
+      .collect()
+  }
+
+  pub async fn validate_operations(&self, operations: Vec<String>) -> Valid<(), String> {
+    match self.to_validation_schema() {
+      Ok(schema) => {
+        let mut execution = vec![];
+
+        for op in operations.iter() {
+          match tokio::fs::read_to_string(op).await {
```

Comment:
we have constraints in some environments like wasm on reading files so it's important to not have IO operations inside the core logic like blueprint. Let's move this reading out of blueprint and move it to `tc.rs` file, here just accept the operation content itself